app/controllers/application_controller.rb

```diff
@@ -1,2 +1,12 @@
 class ApplicationController < ActionController::API
+ def verify_token(data)
+  token = request.env["HTTP_AUTHORIZATION"]
+    if  token && User.decode_token(token)
+   # render json: { message:"you are logged in",info: token}
```

Comment:
It looks like this function isn't doing anything in the case of a valid token (and just delegating to whatever action is the intended destination).

Having an empty `if` is pointless. Maybe this can be written as `render error unless token && User.decode_token` or `if !token || !User.decode_token render error` and leave it at that.